File: Problem/Easy/1701-1800/1790. Check if One String Swap Can Make Strings Equal/_b.java
class Solution {
	public boolean areAlmostEqual(String s1, String s2) {
		int first = -1;
		int second = -1;

		for (int i = 0; i < s1.length(); i++)
			if (s1.charAt(i) != s2.charAt(i))
				if (first == -1)
					first = i;
				else if (second == -1)
					second = i;
				else
					return false;

		if (first == -1)
			return true;

		if (second == -1)
			return false;

		return s1.charAt(first) == s2.charAt(second) &&
				s1.charAt(second) == s2.charAt(first);
	}
}

File: Problem/Easy/1701-1800/1790. Check if One String Swap Can Make Strings Equal/b.py
class Solution:
    def areAlmostEqual(self, s1: str, s2: str) -> bool:
        # keep track of found differences
        first = -1
        second = -1

        for i in range(len(s1)):
            if s1[i] != s2[i]:
                if first == -1:
                    first = i
                elif second == -1:
                    second = i
                else:
                    # found more than two differences
                    return False

        return s1[first] == s2[second] and s1[second] == s2[first]
        # if differences are equal in cross then strings would become equal

        # incase first = -1 or second = -1
        # no need to write extra condition like in java
